CMake improvements, etc

rpavlik opened the pull request:
@h3xx can you take a look and see if this all looks/works right for you? I can build it and run the tests in both static and shared mode on Linux here, and it seems to work fine.

Commit:
Compile the static library with correct ALUT_API

rpavlik committed:
Fix URL

rpavlik committed:
More gitignore

rpavlik committed:
Clean up CMake some more

rpavlik committed:
Format CMake file

rpavlik committed:
Adjust how we specify install dirs

rpavlik committed:
Adjust version properties to match autotools behavior

rpavlik committed:
Add tests to ctest

h3xx commented on CMakeLists.txt:
```diff
@@ -167,69 +181,79 @@ if(POTENTIAL_FLAGS)
-set(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
-set(bindir ${CMAKE_INSTALL_FULL_BINDIR})
-set(includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
```
This regresses a fix in #11.

If one uses an absolute path for this, e.g. `-DCMAKE_INSTALL_LIBDIR=/lib64`, this will be put in the .pc file.

```
libdir=${exec_prefix}//lib65
```

I know it's not DRY or autotools-like but most of the CMake-based projects I encounter regularly install .pc files with full paths in it:

```
prefix=/usr
libdir=/usr/lib64
```

rpavlik replied:
oh, good point, i'll undo that part of my change.

h3xx commented on include/AL/alut.h:
```diff
@@ -17,8 +17,10 @@ extern "C" {
- #if defined (ALUT_BUILD_LIBRARY)
+ #if defined(ALUT_BUILD_LIBRARY)
   #define ALUT_API __declspec(dllexport)
+ #elif defined(ALUT_BUILD_STATIC)
```
I don't see where this macro is set. Grep thru whole project for 'ALUT_BUILD_STATIC' only reveals this spot in the code.

rpavlik replied:
good catch

h3xx commented on src/CMakeLists.txt:
```diff
@@ -29,34 +16,42 @@ set(ALUT_SOURCES
-	${MAJOR_VERSION})
-target_link_libraries(alut ${OPENAL_LIBRARY})
+	PROPERTIES PUBLIC_HEADER "${ALUT_HEADERS}"
+			   VERSION ${MAJOR_VERSION}.${MINOR_VERSION}.${BUILD_VERSION}
```
This regresses a change introduced in 6fdc3652705ac33967062bea27e921969460d4cb.

HOWEVER, the change in *that* commit made it so the Autotools build installed different lib names from CMake.

Your change here shifts it back to them both installing `libalut.so.0.1.0`, which I think we want.

rpavlik replied:
It should also install a symlink libalut.so.0

h3xx reviewed (commented):
The build works for me, but adds a regression for an edge case.

rpavlik committed:
Define ALUT_BUILD_STATIC for static builds and ALUT_BUILD_LIBRARY for dynamic

rpavlik commented:
OK, is this better? I should probably test on Windows I guess.

rpavlik commented:
oh hmm macro redefinition.

rpavlik commented:
OK, tested on Windows, works great.

rpavlik commented:
And with latest fixes it works when building as a static library on Windows too.

rpavlik committed:
Copy DLL on Windows so tests can run

rpavlik committed:
Fix consuming static library on Windows

rpavlik committed:
Update pc file and freealut-config for static library

rpavlik committed:
Fix name and add mailmap file

I have also attempted to correct the git-svn generated attributions
using mailmap, based on data in the AUTHORS file and ChangeLog.

rpavlik committed:
Add self to AUTHORS to clarify my role

rpavlik committed:
Update README with project status.

"Complete" but effectively unmaintained.
